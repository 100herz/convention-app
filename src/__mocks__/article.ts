import { Article } from '@models/article'

export const article: Article = {
  id: 18569,
  date_gmt: '2020-11-11T08:25:41',
  title: {
    rendered: '„New Normal“ bei Leonardo Hotels &#8211; mit neuen MICE-Formaten und Konzepten',
  },
  content: {
    rendered:
      '<p>Durch smarte Raumanpassungen, große Flexibilität bei Buchungen und Stornierungen sowie die Etablierung eines extern zertifizierten und weit über den Standard hinaus gehenden Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen ihrer MICE-Kunden an zukunftsfähigen Veranstaltungsformaten eingestellt. Neue Pausenkonzepte, digitalen Helfer, verlässliche Partner sowie virtuelle und hybride Veranstaltungsformate umfassen die Angebotsvielfalt der Leonardo Hotels.  Diese sind überzeugt, dass persönliche Meetings mit Sicherheit wieder Zukunft haben.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Durch neue Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen an zukunftsfähigen Veranstaltungsformaten eingestellt.</p>\n',
    protected: false,
  },
  featured_media: 18571,
  categories: [6, 1],
  acf: {
    sponsored_by: 'XYZ',
    should_push: false,
    pinned: false,
    featured_slider: false,
    news_slider: true,
  },
  featured_image_thumb:
    'https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg',
  featured_image_medium:
    'https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg',
  categories_names: ['Hotels', 'News'],
}

export const articleWithoutFeatured: Article = {
  id: 18592,
  date_gmt: '2020-11-10T11:00:19',
  title: {
    rendered:
      'Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond',
  },
  content: {
    rendered:
      '<p><a href="https://bit.ly/3iPkUhB">Cvent</a> CEO and Founder, Reggie Aggarwal, delivered an inspiring keynote to more than 7,000 event, marketing, and hospitality professionals at <a href="https://bit.ly/30Unj4A">Cvent CONNECT Europe</a> today. In his opening presentation, Aggarwal urged attendees of the virtual event to seize the digital transformation opportunity in front of them to drive the industry’s recovery in 2021 and beyond.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Cvent CEO Reggie Aggarwal predicts “Golden Age of Events” during his keynote to more than 7,000 virtual attendees at Cvent CONNECT Europe </p>\n',
    protected: false,
  },
  featured_media: null,
  categories: [358, 217, 1],
  acf: {
    sponsored_by: '',
    should_push: false,
    pinned: false,
    featured_slider: false,
    news_slider: true,
  },
  featured_image_thumb: '',
  featured_image_medium: '',
  categories_names: ['Branche', 'Events', 'News'],
}

export const articleWithoutSponsored: Article = {
  id: 18543,
  date_gmt: '2020-11-09T09:46:40',
  title: {
    rendered: 'Sicheres Business und Inspiration für morgen: Koreas neue, einzigartige Veranstaltungsorte',
  },
  content: {
    rendered:
      '<figure id="attachment_18559" aria-describedby="caption-attachment-18559" style="width: 618px" class="wp-caption aligncenter"><img class="wp-image-18559 size-large" src="https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-1200x801.jpg" alt="" width="618" height="413" srcset="https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-1200x801.jpg 1200w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-600x400.jpg 600w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-768x513.jpg 768w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-110x75.jpg 110w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300.jpg 1500w" sizes="(max-width: 618px) 100vw, 618px" /><figcaption id="caption-attachment-18559" class="wp-caption-text">Foto: Pairat Pinijkul / shutterstock.com</figcaption></figure>\n<p>Aufgrund eines exzellenten IKT-Umfelds und seiner kulturellen Attraktivität hat sich Korea als effizientes, innovatives Land für Unternehmen und Kongresse in der MICE-Branche etabliert. Hinsichtlich der Anzahl internationaler Veranstaltungen im Land belegte Korea in den letzten Jahren unter den Mitgliedern der Union der Internationalen Verbände (UIA) den ersten und zweiten Platz. Seit dem Ausbruch von Covid-19 hat die schnelle und gut organisierte Reaktion es dem Land ermöglicht, für die Sicherheit vieler Unternehmen und Konventionen zu sorgen. Mit zahlreichen Support-Programmen unterstützt Korea Verbände und Organisationen bei der Planung und Ausrichtung von Events und baut nach wie vor das Angebot an vielfältigen Eventlocations im Land aus.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Aufgrund eines exzellenten IKT-Umfelds und seiner kulturellen Attraktivität hat sich Korea als effizientes, innovatives Land für Unternehmen und Kongresse in der MICE-Branche etabliert.</p>\n',
    protected: false,
  },
  featured_media: 18559,
  categories: [5, 1, 72, 170],
  acf: {
    // sponsored_by: '100HERZ',
    should_push: false,
    featured_slider: true,
    news_slider: true,
    pinned: true,
  },
  featured_image_thumb:
    'https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-300x300.jpg',
  featured_image_medium:
    'https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-600x400.jpg',
  categories_names: ['Destinationen', 'News', 'Reportagen', 'Sponsored content'],
}
