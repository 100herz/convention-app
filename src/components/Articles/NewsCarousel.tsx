import React from 'react'
import { Dimensions } from 'react-native'
import Carousel from 'react-native-snap-carousel'

import ArticlePreview from '@components/Articles/ArticlePreview'
import { Article } from '@models/article'

interface Props {
  articles: Article[]
}

const NewsCarousel: React.FC<Props> = ({ articles }) => {
  const screenWidth = Dimensions.get('screen').width

  const slideItem = ({ item }: { item: Article }) => <ArticlePreview article={item} hasImage={false} ignoreSponsored />

  return (
    <Carousel
      data={articles}
      renderItem={slideItem}
      sliderHeight={150}
      sliderWidth={screenWidth}
      itemWidth={screenWidth / 1.7}
      enableSnap={false}
      activeSlideAlignment="start"
      inactiveSlideOpacity={1}
      inactiveSlideScale={1}
    />
  )
}

export default NewsCarousel
