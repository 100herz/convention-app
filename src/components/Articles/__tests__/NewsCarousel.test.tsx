import React from 'react'
import { render } from '@testing-library/react-native'

import NewsCarousel from '../NewsCarousel'
import { article, articleWithoutFeatured } from '@__mocks__/article'

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

export const mockedNavigate = jest.fn()
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native')
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useNavigationParam: jest.fn(jest.requireActual('@react-navigation/native').useNavigationParam),
  }
})

describe('<NewsCarousel />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<NewsCarousel articles={[article, articleWithoutFeatured]} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
