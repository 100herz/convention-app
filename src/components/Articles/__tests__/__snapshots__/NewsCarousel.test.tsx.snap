// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsCarousel /> renders correctly 1`] = `
<RCTScrollView
  activeAnimationOptions={null}
  activeAnimationType="timing"
  activeSlideAlignment="start"
  activeSlideOffset={20}
  apparitionDelay={0}
  automaticallyAdjustContentInsets={false}
  autoplay={false}
  autoplayDelay={1000}
  autoplayInterval={3000}
  callbackOffsetMargin={5}
  containerCustomStyle={
    Object {
      "marginLeft": -10,
    }
  }
  contentContainerCustomStyle={Object {}}
  contentContainerStyle={
    Array [
      Object {
        "paddingLeft": 0,
        "paddingRight": 308.8235294117647,
      },
      Object {},
    ]
  }
  data={
    Array [
      Object {
        "acf": Object {
          "featured_slider": false,
          "news_slider": true,
          "pinned": false,
          "should_push": false,
          "sponsored_by": "XYZ",
        },
        "categories": Array [
          6,
          1,
        ],
        "categories_names": Array [
          "Hotels",
          "News",
        ],
        "content": Object {
          "protected": false,
          "rendered": "<p>Durch smarte Raumanpassungen, große Flexibilität bei Buchungen und Stornierungen sowie die Etablierung eines extern zertifizierten und weit über den Standard hinaus gehenden Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen ihrer MICE-Kunden an zukunftsfähigen Veranstaltungsformaten eingestellt. Neue Pausenkonzepte, digitalen Helfer, verlässliche Partner sowie virtuelle und hybride Veranstaltungsformate umfassen die Angebotsvielfalt der Leonardo Hotels.  Diese sind überzeugt, dass persönliche Meetings mit Sicherheit wieder Zukunft haben.</p>
",
        },
        "date_gmt": "2020-11-11T08:25:41",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Durch neue Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen an zukunftsfähigen Veranstaltungsformaten eingestellt.</p>
",
        },
        "featured_image_medium": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg",
        "featured_image_thumb": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg",
        "featured_media": 18571,
        "id": 18569,
        "title": Object {
          "rendered": "„New Normal“ bei Leonardo Hotels &#8211; mit neuen MICE-Formaten und Konzepten",
        },
      },
      Object {
        "acf": Object {
          "featured_slider": false,
          "news_slider": true,
          "pinned": false,
          "should_push": false,
          "sponsored_by": "",
        },
        "categories": Array [
          358,
          217,
          1,
        ],
        "categories_names": Array [
          "Branche",
          "Events",
          "News",
        ],
        "content": Object {
          "protected": false,
          "rendered": "<p><a href=\\"https://bit.ly/3iPkUhB\\">Cvent</a> CEO and Founder, Reggie Aggarwal, delivered an inspiring keynote to more than 7,000 event, marketing, and hospitality professionals at <a href=\\"https://bit.ly/30Unj4A\\">Cvent CONNECT Europe</a> today. In his opening presentation, Aggarwal urged attendees of the virtual event to seize the digital transformation opportunity in front of them to drive the industry’s recovery in 2021 and beyond.</p>
",
        },
        "date_gmt": "2020-11-10T11:00:19",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Cvent CEO Reggie Aggarwal predicts “Golden Age of Events” during his keynote to more than 7,000 virtual attendees at Cvent CONNECT Europe </p>
",
        },
        "featured_image_medium": "",
        "featured_image_thumb": "",
        "featured_media": null,
        "id": 18592,
        "title": Object {
          "rendered": "Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond",
        },
      },
    ]
  }
  decelerationRate="fast"
  directionalLockEnabled={true}
  disableVirtualization={false}
  enableMomentum={false}
  enableSnap={false}
  firstItem={0}
  forwardedRef={[Function]}
  getItem={[Function]}
  getItemCount={[Function]}
  hasParallaxImages={false}
  horizontal={true}
  inactiveSlideOpacity={1}
  inactiveSlideScale={1}
  inactiveSlideShift={0}
  initialNumToRender={7}
  invertStickyHeaders={false}
  inverted={false}
  itemWidth={441.1764705882353}
  keyExtractor={[Function]}
  layout="default"
  lockScrollTimeoutDuration={1000}
  lockScrollWhileSnapping={false}
  loop={false}
  loopClonesPerSide={3}
  maxToRenderPerBatch={15}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderRelease={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchStart={[Function]}
  overScrollMode="never"
  pinchGestureEnabled={false}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={1}
  scrollsToTop={false}
  shouldOptimizeUpdates={true}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  slideStyle={
    Object {
      "flex": 1,
      "marginHorizontal": 10,
    }
  }
  sliderWidth={750}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flexDirection": "row",
      "marginLeft": -10,
      "opacity": 0,
      "width": 750,
    }
  }
  swipeThreshold={20}
  updateCellsBatchingPeriod={50}
  useScrollView={false}
  vertical={false}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={15}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "width": 441.1764705882353,
            },
            Object {
              "flex": 1,
              "marginHorizontal": 10,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "paddingHorizontal": 15,
                "width": 620,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  Array [
                    Object {},
                    Object {},
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  style={
                    Object {
                      "color": "#333333",
                      "fontFamily": "sans",
                      "fontSize": 13,
                    }
                  }
                >
                  „New Normal“ bei Leonardo Hotels – mit neuen MICE-Formaten und Konzepten
                </Text>
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingTop": 5,
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#872232",
                    "fontFamily": "sans",
                    "fontSize": 10,
                    "marginLeft": 5,
                  }
                }
              >
                11. November 2020
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="button"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#872232",
                    "borderRadius": 4,
                    "paddingHorizontal": 15,
                    "paddingVertical": 2,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "sans-bold",
                      "fontSize": 12,
                      "textTransform": "uppercase",
                    }
                  }
                  testID="text-output"
                >
                  Details
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "width": 441.1764705882353,
            },
            Object {
              "flex": 1,
              "marginHorizontal": 10,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "paddingHorizontal": 15,
                "width": 620,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  Array [
                    Object {},
                    Object {},
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  style={
                    Object {
                      "color": "#333333",
                      "fontFamily": "sans",
                      "fontSize": 13,
                    }
                  }
                >
                  Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond
                </Text>
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingTop": 5,
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#872232",
                    "fontFamily": "sans",
                    "fontSize": 10,
                    "marginLeft": 5,
                  }
                }
              >
                10. November 2020
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="button"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#872232",
                    "borderRadius": 4,
                    "paddingHorizontal": 15,
                    "paddingVertical": 2,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "sans-bold",
                      "fontSize": 12,
                      "textTransform": "uppercase",
                    }
                  }
                  testID="text-output"
                >
                  Details
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
